client_src/js/lib/modules/printerSettings.js

```diff
@@ -1,1325 +1,1403 @@
+      document.getElementById("psPowerOffURL").placeholder =
+        currentPrinter.powerSettings.powerOffURL;
+      document.getElementById("psPowerToggleCommand").placeholder =
+        currentPrinter.powerSettings.powerToggleCommand;
+      document.getElementById("psPowerToggleURL").placeholder =
+        currentPrinter.powerSettings.powerToggleURL;
+      document.getElementById("psPowerStateCommand").placeholder =
+        currentPrinter.powerSettings.powerStatusCommand;
+      document.getElementById("psPowerStateURL").placeholder =
+        currentPrinter.powerSettings.powerStatusURL;
+    }
+    document
+      .getElementById("resetPowerFields")
+      .addEventListener("click", async () => {
+        try {
+          await OctoFarmClient.post(
+            "printers/killPowerSettings/" + currentPrinter?._id
+          );
+          UI.createAlert(
+            "success",
+            "Successfully cleared Power Settings",
+            3000,
+            "clicked"
+          );
+        } catch (e) {
+          UI.createAlert(
+            "error",
+            `Unable to clear power settings... Error: ${e}`,
+            3000,
+            "clicked"
+          );
+        }
+      });
+  }
+
+  static async setupAlertsTab(currentPrinter) {
+    let scripts = await OctoFarmClient.get("scripts/get");
+    const printerScripts = [];
+    scripts.alerts.forEach((script) => {
+      if (
+        script.printer === currentPrinter._id ||
+        script.printer.length === 0
+      ) {
+        printerScripts.push({
+          _id: script._id,
+          active: script.active,
+          message: script.message,
+          scriptLocation: script.scriptLocation,
+          trigger: script.trigger,
+        });
+      }
+    });
+
+    const alertsTable = document.getElementById("printerAltersTableBody");
+    alertsTable.innerHTML = "";
+    printerScripts.map(async (s) => {
+      alertsTable.insertAdjacentHTML(
+        "beforeend",
+        `
+            <tr>
+              <td>
+              <form class="was-validated">
+                    <div class="custom-control custom-checkbox mb-3">
+                          <input type="checkbox" class="custom-control-input" id="activePrinter-${s._id}" required="">
+                          <label class="custom-control-label" for="activePrinter-${s._id}"></label>
+                      </div>
+                  </form>
+              </td>
+              <td>
+              <select class="custom-select" id="triggerPrinter-${s._id}" disabled>
+               </select>
+                     </td>
+              <td>${s.scriptLocation} </td>
+              <td>${s.message}</td>
+            </tr>
+
+        `
+      );
+      document.getElementById(`activePrinter-${s._id}`).checked = s.active;
+      const triggerSelect = document.getElementById(`triggerPrinter-${s._id}`);
+      triggerSelect.innerHTML = await Script.alertsDrop();
+      triggerSelect.value = s.trigger;
+    });
```

Comment:
Just trying out .map here rather than for each. Depends what your doing really but this can be faster and more resilient.